pages/jobs/index.js

```diff
@@ -10,11 +10,9 @@ import AnnouncementHero from "../../components/campaigns/AnnoucementHero";
 import Empty from "../../components/illustrations/empty";
 
 export default function JobsIndexPage() {
-  const { navItems } = useContext(JobsContext);
-  navItems.map((job, index) => {
-    if (new Date().getTime() > new Date(job.closingOn).getTime()) {
-      navItems.splice(index, 1);
```

Comment:
`splice` modifies the original array so when it iterates to the next element it might not be there already or it's not the one you expect.